src/mdlComponentHandler.js

```diff
@@ -86,6 +86,15 @@ var componentHandler = (function() {
    * the element to.
    */
   function upgradeElementInternal(element, jsClass) {
+    if (jsClass === undefined) {
+      for (var i = 0; i < registeredComponents_.length; i++) {
```

Comment:
Turns out, `forEach` really is slower than a `for(… in …)` loop, but I find `forEach` so much for expressive that it’s worth the trade-off when talking about time-uncritical code. 

Thinking about whether this is considered time critical, I would actually have to be leaning towards yes and therefore have to agree with @jasonmayes that the `for` loop is probably more appropriate here.